src/jdk.javadoc/share/classes/jdk/javadoc/internal/doclets/toolkit/util/Utils.java

```diff
@@ -2748,14 +2764,125 @@ public DocFinder docFinder() {
+    }
+
+    private class Overrides implements Iterator<ExecutableElement> {
+
+        // prefer java.util.Deque to java.util.Stack API for stacks
+        final Deque<TypeElement> searchStack = new ArrayDeque<>();
+        final Set<TypeElement> visited = new HashSet<>();
+
+        final ExecutableElement overrider;
+        ExecutableElement next;
+
+        public Overrides(ExecutableElement method) {
+            if (method.getKind() != ElementKind.METHOD) {
+                throw new IllegalArgumentException(diagnosticDescriptionOf(method));
+            }
+            overrider = method;
+            // java.lang.Object is to be searched for overrides last
+            searchStack.push(JAVA_LANG_OBJECT);
+            searchStack.push((TypeElement) method.getEnclosingElement());
+        }
+
+        @Override
+        public boolean hasNext() {
+            if (next != null) {
+                return true;
+            }
+            updateNext();
+            return next != null;
+        }
+
+        @Override
+        public ExecutableElement next() {
+            if (!hasNext()) {
+                throw new NoSuchElementException();
+            }
+            var n = next;
+            updateNext();
+            return n;
+        }
+
+        private void updateNext() {
+            while (!searchStack.isEmpty()) {
+                // replace the top class or interface with its supertypes
+                var t = searchStack.pop();
+
+                // <TODO refactor once java.util.List.reversed() from
+                //   SequencedCollection is available>
+                var filteredSupertypes = typeUtils.directSupertypes(t.asType()).stream()
+                        .map(t_ -> (TypeElement) ((DeclaredType) t_).asElement())
+                        // filter out java.lang.Object using the fact that at
+                        // most one class type comes first in the stream of
+                        // direct supertypes
+                        .dropWhile(JAVA_LANG_OBJECT::equals)
+                        .filter(visited::add) // idempotent side effect
+                        .collect(Collectors.toCollection(ArrayList::new));
+                // push supertypes in reverse order, so that they are popped
+                // back in the initial order
+                Collections.reverse(filteredSupertypes);
+                filteredSupertypes.forEach(searchStack::push);
+                // </TODO>
+
+                // consider only the declared methods for consistency with
+                // the existing facilities, such as Utils.overriddenMethod
+                // and VisibleMemberTable.getImplementedMethods
+                TypeElement peek = searchStack.peek();
+                if (peek == null) {
+                    next = null; // end-of-hierarchy
+                    break;
+                }
+                if (isPlainInterface(peek) && !isPublic(peek) && !isLinkable(peek)) {
```

Comment:
What is the significance of the `isPublic` check?
I'm developing a knee-jerk reaction to such checks, in the face of access-related command-line options.

Reply:
That directly mimics the current behaviour at addSuperInterfaces(Utils.java:802), which in turn is used at ImplementedMethods(VisibleMemberTable.java:1064) to collect overridden methods in superinterfaces. If you look around, it's quite an idiom and predates this PR.